src/state/pools/fetchVaultPublic.ts

```diff
@@ -0,0 +1,61 @@
+import BigNumber from 'bignumber.js'
+import { convertSharesToCake } from 'views/Pools/helpers'
+import { getCakeVaultContract } from 'utils/contractHelpers'
+import makeBatchRequest from 'utils/makeBatchRequest'
+
+const cakeVaultContract = getCakeVaultContract()
+
+export const fetchPublicVaultData = async () => {
+  try {
+    const [sharePrice, shares, estimatedCakeBountyReward, totalPendingCakeHarvest] = await makeBatchRequest([
+      cakeVaultContract.methods.getPricePerFullShare().call,
+      cakeVaultContract.methods.totalShares().call,
+      cakeVaultContract.methods.calculateHarvestCakeRewards().call,
+      cakeVaultContract.methods.calculateTotalPendingCakeRewards().call,
+    ])
+    const totalSharesAsBigNumber = new BigNumber(shares as string)
+    const sharePriceAsBigNumber = new BigNumber(sharePrice as string)
+    const totalCakeInVaultEstimate = convertSharesToCake(totalSharesAsBigNumber, sharePriceAsBigNumber)
+    return {
+      totalShares: totalSharesAsBigNumber.toJSON(),
+      pricePerFullShare: sharePriceAsBigNumber.toJSON(),
+      totalCakeInVault: totalCakeInVaultEstimate.cakeAsBigNumber.toJSON(),
+      estimatedCakeBountyReward: new BigNumber(estimatedCakeBountyReward as string).toJSON(),
+      totalPendingCakeHarvest: new BigNumber(totalPendingCakeHarvest as string).toJSON(),
+    }
+  } catch (error) {
+    return {
+      totalShares: null,
+      pricePerFullShare: null,
+      totalCakeInVault: null,
+      estimatedCakeBountyReward: null,
+      totalPendingCakeHarvest: null,
+    }
+  }
+}
+
+export const fetchVaultFees = async () => {
+  try {
+    const [performanceFee, callFee, withdrawalFee, withdrawalFeePeriod] = await makeBatchRequest([
+      cakeVaultContract.methods.performanceFee().call,
+      cakeVaultContract.methods.callFee().call,
+      cakeVaultContract.methods.withdrawFee().call,
+      cakeVaultContract.methods.withdrawFeePeriod().call,
+    ])
+    return {
+      performanceFee: parseInt(performanceFee as string, 10),
```

Comment:
Is `as string` needed here? I guess its because the return type of calls is `any` but it looks like TS doesn't complain if you pass `any` to `parseInt`

Reply:
You could go `as any` but it seems reasonable to type it?

Removing the type all together throws an error:
```
Argument of type 'unknown' is not assignable to parameter of type 'string'
```